src/display.rs

```diff
@@ -162,6 +165,30 @@ impl Display for RawTransaction {
     }
 }
 
+impl Display for NodeInfo {
+    fn to_json(&self) -> Json {
+        let mut info_pair = Map::new();
+        info_pair.insert(
+            String::from("address"),
+            Json::from(format!("0x{}", hex::encode(&self.address))),
+        );
+        let net_info = self.net_info.as_ref().unwrap();
+        info_pair.insert(String::from("origin"), Json::from(net_info.origin));
+        let multi_address: Multiaddr = net_info.multi_address[..].parse().unwrap();
+        for i in multi_address.iter() {
+            let info_string = format!("{}", i);
+            let s: Vec<&str> = info_string.split('/').collect();
+            let _v = match s[1] {
+                "dns4" => info_pair.insert(String::from("host"), Json::from(s[2])),
+                "tcp" => info_pair.insert(String::from("port"), Json::from(s[2])),
+                "tls" => info_pair.insert(String::from("domain"), Json::from(s[2])),
+                _ => panic!("multi address({:?}) can't parse", net_info.multi_address)
+            };
+        }
+        Json::from(info_pair)
```

Comment:
It would be better to use the `json!` macro. [doc](https://docs.rs/serde_json/latest/serde_json/#constructing-json-values)

Reply:
Different network submodule choice need to contain different content in `json!'. So I insert info_pair to a Map and make `Value` from it finally, it satisfies both `p2p` and `tls` situation.